### PR TITLE
Fog improvements

### PR DESCRIPTION
**Pull request by pixelflinger**
- optimized the fog computations
- fixed issues with the skybox
- added the option of getting the fog color from the IBL envmap

### CODE REVIEW

File: shaders/src/fog.fs [hunk 1]
```diff
@@ -2,43 +2,49 @@
...
+        // don't go above requested max opacity
+        fogOpacity = min(fogOpacity, frameUniforms.fogMaxOpacity);
+
+        // compute fog color
+        vec3 fogColor = frameUniforms.fogColor;
+
+        if (frameUniforms.fogColorFromIbl > 0.0) {
+            // get fog color from envmap
+            float lod = frameUniforms.iblRoughnessOneLevel;
+            fogColor = textureLod(light_iblSpecular, view, lod).rgb * frameUniforms.iblLuminance;
```
Comment (romainguy) [inline shaders/src/fog.fs]: `*=` instead of `=`? That way you can tint IBL fog by using a color != 1,1,1.

File: shaders/src/fog.fs [hunk 2]
```diff
@@ -2,43 +2,49 @@
...
+        }
+
+        fogColor *= fogOpacity;
+        if (frameUniforms.fogInscatteringSize >= 0.0) {
+            // compute a new line-integral for a different start distance
+            float inscatteringIntegral = fogIntegralFunctionOfDistance * max(d - frameUniforms.fogInscatteringStart, 0.0);
+            float inscatteringOpacity = max(1.0 - exp2(-inscatteringIntegral), 0.0);
+            //float inscatteringOpacity = clamp(inscatteringIntegral, 0.0, 1.0);
```
Comment (romainguy) [inline shaders/src/fog.fs]: Remove?

File: shaders/src/getters.vs [hunk 3]
```diff
@@ -142,6 +142,9 @@ vec4 computeWorldPosition() {
 #else
     mat4 transform = getWorldFromClipMatrix();
     vec4 position = transform * getPosition();
+    if (abs(position.w) < MEDIUMP_FLT_MIN) {
+        position.w = sign(position.w) != 0.0 ? sign(position.w) * MEDIUMP_FLT_MIN : MEDIUMP_FLT_MIN;
```
Comment (romainguy) [inline shaders/src/getters.vs]: Why the test here? If you're going to call sign() every time, might as well do sign(w) * FLT_MIN as the expression. Even better just write:

position.w = position.w < 0.0 ? -FLT_MIN : FLT_MIN

File: shaders/src/fog.fs (hunk 1 above)
Comment (pixelflinger) [reply shaders/src/fog.fs]: yeah I was wondering about that.

File: shaders/src/fog.fs (hunk 2 above)
Comment (pixelflinger) [reply shaders/src/fog.fs]: I wanted to keep it for now until I'm really sure. After reading more articles about this, I still don't see anyone doing this (AFAIK), which is strange.

Comment (romainguy) [reply shaders/src/fog.fs]: It seems easy to do and worth it. It's an extra feature at super low cost and could be useful to simulate time of day in some cases.

**Pull request merged**